src/app/config.ts

```diff
@@ -0,0 +1,25 @@
+import { cleanEnv, url, str } from "envalid";
+
+type Config = {
+  auth: {
+    url: string;
+    realm: string;
+    clientId: string;
+  };
+}
+
+const env = cleanEnv(import.meta.env, {
+  VITE_KEYCLOAK_URL: url(),
+  VITE_KEYCLOAK_REALM: str(),
+  VITE_KEYCLOAK_CLIENT_ID: str(),
+});
+
+const config: Config = {
+  auth: {
+    url: env.VITE_KEYCLOAK_URL,
+    realm: env.VITE_KEYCLOAK_REALM,
+    clientId: env.VITE_KEYCLOAK_CLIENT_ID,
+  }
+}
```

Comment:
We probably don't need to refer the Config type anywhere, so it can be removed. And, if we do, you can always add `type Config = typeof config;`. Also, `as const` should be added to create config as read-only.

```suggestion
const env = cleanEnv(import.meta.env, {
  VITE_KEYCLOAK_URL: url(),
  VITE_KEYCLOAK_REALM: str(),
  VITE_KEYCLOAK_CLIENT_ID: str(),
});

const config = {
  auth: {
    url: env.VITE_KEYCLOAK_URL,
    realm: env.VITE_KEYCLOAK_REALM,
    clientId: env.VITE_KEYCLOAK_CLIENT_ID,
  }
} as const;
```

Reply:
Good point. Fixed as suggested. Damn copy paste